integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/scaletest/QuerySpecs.scala

```diff
@@ -47,57 +48,44 @@ class QuerySpecs(config: Config, spark: SparkSession) {
 
   /**
    * To get numeric columns in a dataframe.
-   * Now numeric columns are limited to [byte, int, long, decimal]
+   * Currently numeric columns are limited to [byte, int, long, decimal]
    *
    * @param table table name
    * @return numeric type column names
    */
   private def getNumericColumns(table: String): Seq[String] = {
-    assert(tables.contains(table), s"Invalid table: $table, candidate tables are ${
-      tables
-        .mkString(",")
-    }")
+    assert(tables.contains(table), s"Invalid table: $table, candidate tables " +
+        s"are ${tables.mkString(",")}")
     val df = spark.read.format(config.format).load(s"$baseInputPath/$table")
-    // Get the data types of all columns
-    val columnDataTypes = df.dtypes
-
-    // Filter columns that are numeric
-    val numericColumns = columnDataTypes.filter {
-      case (_, dataType) =>
-        dataType == "ByteType" || dataType == "IntegerType" || dataType == "LongType" || dataType
-          .startsWith("DecimalType")
-    }.map {
-      case (columnName, _) => columnName
+    df.schema.map { field =>
+      (field.name, field.dataType)
+    }.collect {
+      case (columnName, ByteType | IntegerType | LongType | _: DecimalType) =>
+        columnName
```

Comment:
no need to fix, just a remark that even with this improvement map is not really necessary (unless it causes issues with different Spark versions) if we  collect with 
```scala
case StructField(columnName, ByteType | IntegerType | LongType | _: DecimalType, _, _) => columnName
```